modules/core/src/main/scala/fs2/kafka/internal/WithConsumer.scala

```diff
@@ -7,12 +7,14 @@
 package fs2.kafka.internal
 
 import cats.effect.{Blocker, Concurrent, ContextShift, Resource}
+import cats.effect.concurrent.Semaphore
 import cats.implicits._
 import fs2.kafka.{KafkaByteConsumer, ConsumerSettings}
 import fs2.kafka.internal.syntax._
 
 private[kafka] sealed abstract class WithConsumer[F[_]] {
   def apply[A](f: KafkaByteConsumer => F[A]): F[A]
+  def blocking[A](f: KafkaByteConsumer => A): F[A]
```

Comment:
I'm slightly 👎 against `blocking`, since it's somewhat easy to return `F[A]` here and discard it by accident. Although if we still want it, I'd suggest we add it to `WithProducer` as well to keep them consistent.

Reply:
This is mandated by the design of cats-effect pretty much. Without it, you will have to manually thread an `EC` for blocking op, whereas `Blocker` disappears with this choice

Reply:
You *could* do `Sync[F].blocking` at the use site, no?

Reply:
oh yeah sure

Reply:
I still prefer `blocking` though. I think the errors Viktor describes will be much easier to catch (either with the compiler or tests) than an error with using `delay` instead of `blocking` in calling code. (And it's less boilerplatey too)

Reply:
I _think_ all calls are blocking, so no need to differentiate between the two. With CE3 we would just merge the two into the signature of `blocking`, so maybe do that here instead then (and keep it as `apply`)?

Reply:
You're right we should do the same thing with `WithProducer` - I'll make that change. There's also uses in `KafkaConsumerActor` I still need to change.

Unfortunately I don't think we can completely get rid of the current `apply` in `WithProducer`, because of the way [TransactionalKafkaProducer uses `bracket`](https://github.com/fd4s/fs2-kafka/blob/114136ad0fe9063fd799d75c3c938d78f63013ec/modules/core/src/main/scala/fs2/kafka/TransactionalKafkaProducer.scala#L70-L91). Maybe we should rename `apply` to `defer` (the name for `suspend` in CE3) though.